### PR TITLE
Feature/12811 instructions to the top up

### PR DESCRIPTION
**Pull request by alxtz**
# Issue
https://github.com/Xfers/Xfers/issues/12811

image

image

# Test cases
- [x] provides an instruction page when user has no existing cookie
- [x] writes into cookie when clicked into I got it & checkbox is checked
- [x] add two helper buttons into top up fund source page
- [x] add *I need more information* link with helper buttons (design https://www.figma.com/file/wEBaKJY5abdhyxPCYhafJw/xfers_popup_update?node-id=0%3A1
- [x] when exiting the instruction page for the first time, don't show checkbox in this flow again

# Known issues
- this needs to be merged first to use the icons & components https://github.com/Xfers/Xfers/pulls

### CODE REVIEW

File: JavaScript/src/components/flows/WalletFlow/actions/stepperActions.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { SHOW_INSTRUCTION_PAGE, HIDE_INSTRUCTION_PAGE } from './constants'
+
+export const hideInstructionPage = () => {
```
Comment (lfkeong) [inline JavaScript/src/components/flows/WalletFlow/actions/stepperActions.js]:
```
export const hideInstructionPage = () => ({
    type: HIDE_INSTRUCTION_PAGE
})

```

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/ListItem.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+
+import { View, TwoColsRowBar, SvgIcon } from '@xfers/xfers-components'
+
+export default function ListItem({ children, icon, checked }) {
```
Comment (lfkeong) [inline JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/ListItem.js]: You're not using `checked` yea?

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/ListItem.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        noSidePadding
+        leftColProps={{
+          size: { sm: 3, md: 2 },
+          content: (
+            <View height="35px" width="35px" paddingTop="0px">
+              {checked ? (
+                <SvgIcon icon="success" xfersColor="blue" />
+              ) : (
+                <SvgIcon icon={icon} xfersColor="gradient" />
+              )}
```
Comment (lfkeong) [inline JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/ListItem.js]: Do we need to do conditions here?

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/TopUp.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import React, { Component } from 'react'
+import cookies from 'js-cookie'
```
Comment (lfkeong) [inline JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/TopUp.js]: Looks like this is not used in this file, can remove

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/index.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+                <span style={{ marginRight: '5px' }}>
+                  I need more information
+                </span>
+                <SvgIcon icon="link" xfersColor="blue" />
+              </a>
+            </View>
+          </ListItem>
+
+          {this.props.showCheckboxInInstructionPage && (
+            <div style={{ display: 'flex' }}>
```
Comment (lfkeong) [inline JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/index.js]: <View>

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/index.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+              <div style={{ transform: 'scale(1.2)' }}>
+                <Checkbox
+                  checked={this.state.isChecked}
+                  onChange={isChecked => this.setState({ isChecked })}
+                />
+              </div>
+              <span
```
Comment (lfkeong) [inline JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/index.js]: `<Text></Text>`

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/HelpButtons.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        <div className={styles.iconColumn}>
+          <SvgIcon icon="dialog" width={40} original />
+        </div>
+        <div className={styles.textColumn}>
+          <Text type="label">FAQ</Text>
+          <Text type="boldValue">Having problems?</Text>
+          <Text>Check out our FAQ</Text>
+        </div>
+      </a>
+    </div>
```
Comment (lfkeong) [inline JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/HelpButtons.js]: Can replace `<div></div>` with `<View></View>`. This is for code consistency/predictability purpose.
And `<a></a>` with `<AnchorLink></AnchorLink>`

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/index.js [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+          <ListItem icon="document">
+            <View>
+              <Text type="label">STEP 3</Text>
+              <Text type="boldValue">Wait and check after 5 minutes</Text>
+              <Text fontSize="12px">
+                Successful top-ups should show up in your wallet if all steps
+                are followed.
+              </Text>
+
+              <a
```
Comment (lfkeong) [inline JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/index.js]: `<AnchorLink></AnchorLink>`

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/index.js [hunk 9]
```diff
@@ -36,9 +38,9 @@ export default class TopUpFundSource extends Component {
     return (
       <StickyPanel locale={locale} showBrand>
         {params.flowType === 'payment' ? (
-          <ModalHeader onBack={goBack} spHeader title="Make Payment" />
+          <ModalHeader onClose={closeModal} spHeader title="Make Payment" />
         ) : (
-          <ModalHeader onBack={goBack} spHeader title="Make a Top-up" />
+          <ModalHeader onClose={goBack} spHeader title="Make a Top-up" />
```
Comment (lfkeong) [inline JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/index.js]: `closeModal`, not `goBack`

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/index.js [hunk 10]
```diff
@@ -36,9 +38,9 @@ export default class TopUpFundSource extends Component {
     return (
       <StickyPanel locale={locale} showBrand>
         {params.flowType === 'payment' ? (
-          <ModalHeader onBack={goBack} spHeader title="Make Payment" />
+          <ModalHeader onClose={closeModal} spHeader title="Make Payment" />
```
Comment (lfkeong) [inline JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/index.js]: Did you check if the `closeModal` function is extracted from the props?

File: JavaScript/src/components/flows/WalletFlow/actions/stepperActions.js (hunk 1 above)
Comment (alxtz) [reply JavaScript/src/components/flows/WalletFlow/actions/stepperActions.js]: fixed

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/ListItem.js (hunk 2 above)
Comment (alxtz) [reply JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/ListItem.js]: fixed

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/ListItem.js (hunk 3 above)
Comment (alxtz) [reply JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/ListItem.js]: fixed

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/index.js (hunk 5 above)
Comment (alxtz) [reply JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/index.js]: fixed

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/TopUp.js (hunk 4 above)
Comment (alxtz) [reply JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/TopUp.js]: fixed

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/index.js (hunk 6 above)
Comment (alxtz) [reply JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/index.js]: fixed

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/HelpButtons.js (hunk 7 above)
Comment (alxtz) [reply JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/HelpButtons.js]: fixed

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/index.js (hunk 8 above)
Comment (alxtz) [reply JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/InstructionsPage/index.js]: fixed

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/index.js (hunk 9 above)
Comment (alxtz) [reply JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/index.js]: this is not modified by me but prettier

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/index.js (hunk 10 above)
Comment (alxtz) [reply JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/index.js]: fixed

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/index.js (hunk 9 above)
Comment (alxtz) [reply JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/index.js]: fixed

File: JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/index.js (hunk 10 above)
Comment (alxtz) [reply JavaScript/src/components/screens/Wallet/components/PaymentAndTopUp/components/TopUpFundSource/index.js]: fixed

**Pull request merged**